tests/test_git_init.py

```diff
@@ -0,0 +1,73 @@
+"""Checks that the git repo initialisation works."""
+
+import pathlib
+import shlex
+import subprocess
+
+
+def test_git_not_initialised(
+    tmp_path: pathlib.Path,
+    project_config: dict,
+) -> None:
+    """
+    Checks for the case without git initialisation.
+
+    Args:
+        tmp_path: A temporary directory path object which is unique.
+        project_config: A dictionary with values for the cookiecutter template,
+            as defined in the cookiecutter.json
+    """
+    # Run cookiecutter with `initialise_git_repository=False`
+    subprocess.run(
+        shlex.split(  # noqa: S603
+            f"cookiecutter . --no-input --output-dir {tmp_path} project_name="
+            f"'{project_config['project_name']}' initialise_git_repository=False"
+        ),
+        check=True,
+    )
+
+    test_project_dir = tmp_path / project_config["expected_repo_name"]
+
+    result = subprocess.run(
+        shlex.split(f"git -C {test_project_dir} status"),  # noqa: S603
+        capture_output=True,
+        check=False,
+        text=True,
+    )
+    assert (
+        result.stderr
+        == "fatal: not a git repository (or any of the parent directories): .git\n"
+    ), f"Error: {result.stderr!r}"
+
+
+def test_git_initialised(
```

Comment:
This is a bit WET. Is the DRY version too fiddly?

```python
@pytest.mark.parametrize(
    "git_init_cookiecutter_option", [
        "initialise_git_repository=True", 
       "initialise_git_repository=False"
    ]
)
def test_initialisation_of_git_repo(git_init_cookiecutter_option):
    ...
    # call
    f"'{project_config['project_name']}' {git_init_cookiecutter_option}"
    ...
    # check
    if "True" in git_init_cookiecutter_option:
        assert result.returncode == 0
    else: # should not have found repo
        assert result.stderr == "flibble"
```

Reply:
Good idea, that was pretty moist